src/ERC4973.sol

```diff
@@ -11,41 +11,85 @@ import {ERC721URIStorage} from
 import {BitMaps} from "@openzeppelin/contracts/utils/structs/BitMaps.sol";
 
 import {IERC4973} from "./interfaces/IERC4973.sol";
+import {IERC5192} from "./interfaces/IERC5192.sol";
 
 bytes32 constant AGREEMENT_HASH =
   keccak256("Agreement(address active,address passive,bytes metadata)");
 
 /// @notice Reference implementation of EIP-4973 tokens.
 /// @author Tim Daubenschütz, Rahul Rumalla (https://github.com/rugpullindex/ERC4973/blob/master/src/ERC4973.sol)
-abstract contract ERC4973 is EIP712, ERC721, ERC721URIStorage, IERC4973 {
+abstract contract ERC4973 is
+  EIP712,
+  ERC721,
+  ERC721URIStorage,
+  IERC4973,
+  IERC5192
+{
   using BitMaps for BitMaps.BitMap;
 
   BitMaps.BitMap private _usedHashes;
 
+  modifier notSupported() {
+    revert("notSupported: Soulbound NFT EIP-5192");
```

Comment:
I would change this message in `notSupported: Locked NFT EIP-5192`

Reply:
I‘m now using an error https://github.com/attestate/ERC5192/blob/1ed017b942c6bad4706b62ba84f6db90a041b304/src/ERC5192.sol#L21

Reply:
Good.